src/app/shared/directives/highlight-creation-date.directive.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Directive, ElementRef } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { HighlightCreationDateDirective } from './highlight-creation-date.directive';
+
+
+describe('HighlightCreationDateDirective', () => {
+
+  @Component({
+    template: `
+      <div [appHighlightCreationDate]="creationDate" class="courseItemContainer">
+        <div class="courseItemUpperBlock">
+            <p class="courseItemCreationDate"></p>
+        </div>
+      </div>
+    `
+  })
+  class TestComponent {
+    creationDate: Date = new Date(2023, 5, 30);
+  }
+
+  const elementRefMock = jasmine.createSpyObj('ElementRef', ['nativeElement']);
+
+  let fixture: ComponentFixture<TestComponent>;
+
+  beforeEach( () => {
+    fixture = TestBed.configureTestingModule({
+      declarations: [TestComponent, HighlightCreationDateDirective]
+    }).createComponent(TestComponent);
+
+    fixture.detectChanges();
+  });
+
+  it('should create an instance', () => {
```

Comment:
It you have time you may try to add some more tests for the directive
The documentation is written pretty clear
[testing-attribute-directives](https://angular.io/guide/testing-attribute-directives)

Reply:
Shorty you create the test component
`@Component({
  template: `<div [appHighlightCreationDate]="date"></div>`
})
class TestComponent {
  date: Date;
}`

Then put it to the declarations and create **fixture** from the test component `fixture = TestBed.createComponent(TestComponent);`

And then you can try something like this 

`component.date = new Date();
    component.date.setDate(component.date.getDate() + 15);
    fixture.detectChanges();

    expect(fixture.nativeElement.firstChild.style.borderColor).toBe('blue');`

Reply:
Thank you for detailed comment.
Have done additional tests for custom directive. it was not so scary as it looks at first sight